nipoppy_cli/nipoppy/workflows/pipeline.py

```diff
@@ -270,20 +270,22 @@ def set_up_bids_db(
         """Set up the BIDS database."""
         dpath_bids_db: Path = Path(dpath_bids_db)
 
+        pybids_ignore_patterns = self.pybids_ignore_patterns.copy()
+
         if participant_id is not None:
             add_pybids_ignore_patterns(
-                current=self.pybids_ignore_patterns,
+                current=pybids_ignore_patterns,
```

Comment:
so this function updates things in place, without returning anything?

Reply:
If i understand this correctly, it keeps growing the ignore list by adding (supposedly) completed participants?

Reply:
That's what it used to do (which caused the bug), but it doesn't do that anymore because the list is copied